Bug: Module constants not resolved via 'using' directive

## Description

When using the `using` directive to access constants from a local module, the typechecker incorrectly reports E4001 (undefined variable) errors. Functions from modules work correctly via `using` (fixed in #671), but constants do not.

## Steps to Reproduce

1. Create a module with a constant:
```ez
// libA.ez
module libA
const VERSION string = "1.0.0"
```

2. Import and use the module:
```ez
import & use "./libA"

do main() {
    temp ver string = VERSION  // E4001: undefined variable 'VERSION'
    println(ver)
}
```

## Expected Behavior

The constant should be accessible without the module prefix when `using` is active.

## Actual Behavior

```
error[E4001]: undefined variable 'VERSION'
```

## Root Cause

The typechecker tracks module functions via `moduleFunctions` and module types via `moduleTypes`, but there is no `moduleVariables` map to track module-level constants.

## Proposed Fix

1. Add `moduleVariables map[string]map[string]string` to TypeChecker (module name -> var name -> type)
2. Add `RegisterModuleVariable(moduleName, varName, typeName string)` method
3. Update `isKnownIdentifier` to check `moduleVariables` for `using` modules
4. Update module loading to register module constants

## Affected Tests

- `integration-tests/pass/multi-file/imports/` (uses `VERSION` constant)
- `integration-tests/pass/multi-file/foreach-namespaced-collection/` (uses `lib.Numbers` constant in for_each)

## Related

- #671 - Fixed module functions via `using`
- Part of typechecker hardening effort

--- a/cmd/ez/main.go
+++ b/cmd/ez/main.go
@@ -234,6 +234,8 @@ func checkFile(filename string) {
 	moduleSignatures := make(map[string]map[string]*typechecker.FunctionSignature)
 	// Store module type definitions: alias/moduleName -> typeName -> type
 	moduleTypes := make(map[string]map[string]*typechecker.Type)
+	// Store module variable/constant definitions: alias/moduleName -> varName -> type (#677)
+	moduleVariables := make(map[string]map[string]string)
 
 	for len(modulesToCheck) > 0 {
 		// Pop first module
@@ -334,6 +336,9 @@ func checkFile(filename string) {
 				if moduleTypes[moduleName] == nil {
 					moduleTypes[moduleName] = make(map[string]*typechecker.Type)
 				}
+				if moduleVariables[moduleName] == nil {
+					moduleVariables[moduleName] = make(map[string]string)
+				}
 				// Copy function signatures from this file's typechecker
 				for funcName, sig := range fileTc.GetFunctions() {
 					moduleSignatures[moduleName][funcName] = sig
@@ -344,6 +349,10 @@ func checkFile(filename string) {
 					if t.Kind == typechecker.StructType || t.Kind == typechecker.EnumType {
 						moduleTypes[moduleName][typeName] = t
 					}
+				}
+				// Copy variable/constant definitions from this file's typechecker (#677)
+				for varName, varType := range fileTc.GetVariables() {
+					moduleVariables[moduleName][varName] = varType
 				}
 			}
 
@@ -376,6 +385,13 @@ func checkFile(filename string) {
 	for moduleName, types := range moduleTypes {
 		for typeName, t := range types {
 			tc.RegisterModuleType(moduleName, typeName, t)
+		}
+	}
+
+	// Register all module variable/constant definitions for cross-module type checking (#677)
+	for moduleName, vars := range moduleVariables {
+		for varName, varType := range vars {
+			tc.RegisterModuleVariable(moduleName, varName, varType)
 		}
 	}
 
@@ -770,6 +786,7 @@ func runFile(filename string) {
 
 	moduleSignatures := make(map[string]map[string]*typechecker.FunctionSignature)
 	moduleTypes := make(map[string]map[string]*typechecker.Type)
+	moduleVariables := make(map[string]map[string]string) // (#677)
 	checked := make(map[string]bool)
 	checked[absPath] = true
 
@@ -825,6 +842,9 @@ func runFile(filename string) {
 				if moduleTypes[moduleName] == nil {
 					moduleTypes[moduleName] = make(map[string]*typechecker.Type)
 				}
+				if moduleVariables[moduleName] == nil {
+					moduleVariables[moduleName] = make(map[string]string)
+				}
 				for funcName, sig := range fileTc.GetFunctions() {
 					moduleSignatures[moduleName][funcName] = sig
 				}
@@ -832,6 +852,10 @@ func runFile(filename string) {
 					if t.Kind == typechecker.StructType || t.Kind == typechecker.EnumType {
 						moduleTypes[moduleName][typeName] = t
 					}
+				}
+				// Copy variable/constant definitions (#677)
+				for varName, varType := range fileTc.GetVariables() {
+					moduleVariables[moduleName][varName] = varType
 				}
 			}
 
@@ -852,6 +876,12 @@ func runFile(filename string) {
 	for moduleName, types := range moduleTypes {
 		for typeName, t := range types {
 			tc.RegisterModuleType(moduleName, typeName, t)
+		}
+	}
+	// Register module variables/constants (#677)
+	for moduleName, vars := range moduleVariables {
+		for varName, varType := range vars {
+			tc.RegisterModuleVariable(moduleName, varName, varType)
 		}
 	}
 

--- a/pkg/typechecker/typechecker.go
+++ b/pkg/typechecker/typechecker.go
@@ -158,6 +158,7 @@ type TypeChecker struct {
 	fileUsingModules map[string]bool                          // File-level using modules
 	moduleFunctions  map[string]map[string]*FunctionSignature // Module name -> function name -> signature
 	moduleTypes      map[string]map[string]*Type              // Module name -> type name -> type
+	moduleVariables  map[string]map[string]string             // Module name -> variable name -> type (#677)
 	currentScope     *Scope                                   // Current scope for local variable tracking
 	errors           *errors.EZErrorList
 	source           string
@@ -176,6 +177,7 @@ func NewTypeChecker(source, filename string) *TypeChecker {
 		fileUsingModules: make(map[string]bool),
 		moduleFunctions:  make(map[string]map[string]*FunctionSignature),
 		moduleTypes:      make(map[string]map[string]*Type),
+		moduleVariables:  make(map[string]map[string]string),
 		errors:           errors.NewErrorList(),
 		source:           source,
 		filename:         filename,
@@ -209,6 +211,14 @@ func (tc *TypeChecker) RegisterModuleType(moduleName, typeName string, t *Type) 
 	tc.moduleTypes[moduleName][typeName] = t
 }
 
+// RegisterModuleVariable registers a variable/constant from an imported module (#677)
+func (tc *TypeChecker) RegisterModuleVariable(moduleName, varName, typeName string) {
+	if tc.moduleVariables[moduleName] == nil {
+		tc.moduleVariables[moduleName] = make(map[string]string)
+	}
+	tc.moduleVariables[moduleName][varName] = typeName
+}
+
 // GetModuleFunction retrieves a function signature from a module
 func (tc *TypeChecker) GetModuleFunction(moduleName, funcName string) (*FunctionSignature, bool) {
 	if funcs, ok := tc.moduleFunctions[moduleName]; ok {
@@ -216,6 +226,15 @@ func (tc *TypeChecker) GetModuleFunction(moduleName, funcName string) (*Function
 		return sig, exists
 	}
 	return nil, false
+}
+
+// GetModuleVariable retrieves a variable type from a module (#677)
+func (tc *TypeChecker) GetModuleVariable(moduleName, varName string) (string, bool) {
+	if vars, ok := tc.moduleVariables[moduleName]; ok {
+		typeName, exists := vars[varName]
+		return typeName, exists
+	}
+	return "", false
 }
 
 // GetFunctions returns the functions map (for extracting signatures from module typechecker)
@@ -226,6 +245,11 @@ func (tc *TypeChecker) GetFunctions() map[string]*FunctionSignature {
 // GetTypes returns the types map (for extracting types from module typechecker)
 func (tc *TypeChecker) GetTypes() map[string]*Type {
 	return tc.types
+}
+
+// GetVariables returns the variables map (for extracting constants from module typechecker) (#677)
+func (tc *TypeChecker) GetVariables() map[string]string {
+	return tc.variables
 }
 
 // registerBuiltinTypes adds all built-in types to the registry
@@ -3619,6 +3643,23 @@ func (tc *TypeChecker) isKnownIdentifier(name string) bool {
 			}
 		}
 	}
+	// Check if it's a variable/constant from a user module accessible via 'using' (#677)
+	for moduleName := range tc.fileUsingModules {
+		if vars, ok := tc.moduleVariables[moduleName]; ok {
+			if _, exists := vars[name]; exists {
+				return true
+			}
+		}
+	}
+	if tc.currentScope != nil {
+		for moduleName := range tc.currentScope.usingModules {
+			if vars, ok := tc.moduleVariables[moduleName]; ok {
+				if _, exists := vars[name]; exists {
+					return true
+				}
+			}
+		}
+	}
 	return false
 }
 
@@ -4210,6 +4251,15 @@ func (tc *TypeChecker) inferMemberType(member *ast.MemberExpression) (string, bo
 			}
 			// Return the enum type name
 			return label.Value, true
+		}
+
+		// Check if accessing module variable (e.g., lib.Numbers) (#677)
+		moduleName := label.Value
+		if tc.modules[moduleName] {
+			memberName := member.Member.Value
+			if varType, ok := tc.GetModuleVariable(moduleName, memberName); ok {
+				return varType, true
+			}
 		}
 	}
 

